tests/test_agent.py

```diff
@@ -3886,6 +3886,24 @@ async def test_context_manager():
             assert server2.is_running
 
 
+async def test_implicit_context_manager():
```

Comment:
This is going to fix https://github.com/pydantic/pydantic-ai/issues/2355 as well, can you please add the test from my comment there?

Reply:
Will do